modules/miner/update.go

```diff
@@ -144,5 +253,5 @@ func (m *Miner) ReceiveUpdatedUnconfirmedTransactions(diff *modules.TransactionP
 
 	m.deleteReverts(diff)
 	m.addNewTxns(diff)
-	m.pickNewTransactions(diff)
+	m.adjustUnsolvedBlock()
 }
```

Comment:
Overall this file is looking a lot cleaner from the first pass, code quality and organization is definitely much higher.